repos/system_upgrade/el7toel8/actors/satellite_upgrader/actor.py

```diff
@@ -32,3 +32,10 @@ def process(self):
             api.current_logger().error(
                 'Could not run the installer, please inspect the logs in /var/log/foreman-installer!'
             )
+
+        if facts.postgresql.local_postgresql:
+            api.current_actor().show_message('Re-indexing the database. This can take a while.')
```

Comment:
@evgeni I am not sure whether this will have the effect you want. The process is running in the background and user does not have to see it in the console output (also, will see it only if they see the real console output, nor ssh, etc..). You can keep it, but I suggest to create debug log and also a pre-upgrade report informing people that during the first boot into the upgraded system, this operation will be performed and could take a while, so they will know that even if they log into the system, the upgrade does not have to be finished yet and they need to wait for leapp process to be finished completely.

Reply:
Yeah, the messaging to the user can certainly be improved. For now, I just copied what we already had for the Installer run. Will have a look at the report stuff as you suggest